src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpoint.java

```diff
@@ -141,24 +147,205 @@
+  public ResponseEntity<?> patchCollectionExerciseName(
+          @PathVariable("id") final UUID id,
+          final @RequestBody String name)
+          throws CTPException {
+    log.info("Updating collection exercise {}, setting name to {}", id, name);
+    CollectionExerciseDTO collexDto = new CollectionExerciseDTO();
+    collexDto.setName(name);
+
+    return patchCollectionExercise(id, collexDto);
+  }
+
+  /**
+   * PUT request to update a collection exercise userDescription
+   * @param id Collection exercise Id to update
+   * @param userDescription new value for user description
+   * @throws CTPException on resource not found
+   * @return 200 if all is ok, 400 for bad request, 409 for conflict
+   */
+  @RequestMapping(value = "/{id}/userDescription", method = RequestMethod.PUT, consumes = "text/plain")
+  public ResponseEntity<?> patchCollectionExerciseUserDescription(
+          @PathVariable("id") final UUID id,
+          final @RequestBody String userDescription)
+          throws CTPException {
+    log.info("Updating collection exercise {}, setting userDescription to {}", id, userDescription);
+    CollectionExerciseDTO collexDto = new CollectionExerciseDTO();
+    collexDto.setUserDescription(userDescription);
+
+    return patchCollectionExercise(id, collexDto);
+  }
+
+  /**
+   * PUT request to update a collection exercise userDescription
+   * @param id Collection exercise Id to update
+   * @throws CTPException on resource not found
+   * @return 200 if all is ok, 400 for bad request, 409 for conflict
+   */
+  @RequestMapping(value = "/{id}/surveyId", method = RequestMethod.PUT, consumes = "text/plain")
+  public ResponseEntity<?> patchCollectionExerciseSurveyId(
+          @PathVariable("id") final UUID id,
+          final @RequestBody String surveyId)
+          throws CTPException {
+    log.info("Updating collection exercise {}, setting surveyId to {}", id, surveyId);
+    try {
+      UUID.fromString(surveyId);
+    } catch (IllegalArgumentException e) {
+      throw new CTPException(CTPException.Fault.BAD_REQUEST, e);
+    }
+
+    CollectionExerciseDTO collexDto = new CollectionExerciseDTO();
+    collexDto.setSurveyId(surveyId);
+
+    return patchCollectionExercise(id, collexDto);
+  }
+
+  /**
+   * POST request to create a collection exercise
+   * @param collex A dto containing the data about the collection exercise
+   * @return 201 if all is ok, 400 for bad request, 409 for conflict
+   * @throws CTPException on resource not found
+   */
+  @RequestMapping(method = RequestMethod.POST)
+  public ResponseEntity<?> createCollectionExercise(
+          final @Validated(CollectionExerciseDTO.PostValidation.class) @RequestBody CollectionExerciseDTO collex)
+          throws CTPException {
+    log.info("Creating collection exercise");
+    String surveyId = collex.getSurveyId();
+
+    if (StringUtils.isBlank(surveyId)) {
```

Comment:
Is this covered by the validation in https://github.com/ONSdigital/rm-collectionexercisesvc-api/pull/4/files#diff-896f7d186ede01b90cee7e3076299b31R29

Reply:
When you wrote the comment, yes this is covered by the validation so unnecessary.  However the code has evolved since then (survey can be specified either as a UUID or a business friendly ref so this check exists in the newer code and is necessary)